src/main/java/Echo/TaskManager.java

```diff
@@ -0,0 +1,333 @@
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + ". " + tasks.get(i));
+            }
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Marks a task as done based on the user input.
+     *
+     * @param tokens The user input tokens.
+     */
+    public void markTask(String[] tokens) {
+        if (tokens.length == 2) {
+            int index = Integer.parseInt(tokens[1]);
+            if (isValidIndex(index)) {
+                tasks.get(index - 1).markAsDone();
+                System.out.println("____________________________________________________________");
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  " + tasks.get(index - 1));
+                System.out.println("____________________________________________________________");
+            } else {
+                System.out.println("Invalid task index.");
+            }
+        } else {
+            System.out.println("Invalid command. Usage: mark <index>");
+        }
+        saveTasksToFile();
+    }
+
+    /**
+     * Marks a task as undone based on the user input.
+     *
+     * @param tokens The user input tokens.
+     */
+    public void unmarkTask(String[] tokens) {
+        if (tokens.length == 2) {
+            int index = Integer.parseInt(tokens[1]);
+            if (isValidIndex(index)) {
+                tasks.get(index - 1).markAsUndone();
+                System.out.println("____________________________________________________________");
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println("  " + tasks.get(index - 1));
+                System.out.println("____________________________________________________________");
+            } else {
+                System.out.println("Invalid task index.");
+            }
+        } else {
+            System.out.println("Invalid command. Usage: unmark <index>");
+        }
+        saveTasksToFile();
+    }
+
+    /**
+     * Adds a task based on the user input.
+     *
+     * @param tokens The user input tokens.
+     */
+    public void addTask(String[] tokens) {
+        try {
+            if (tokens.length != 2) {
+                throw new IllegalArgumentException("NO! I don't know what is this! " +
+                        "Invalid command. Supported tasks: todo, deadline, event");
+            }
+            String[] taskTokens = tokens;
+
+            String taskType = taskTokens[0].toLowerCase();
+            if (taskTokens[1].isEmpty()) {
+                throw new IllegalArgumentException("NO! " +
+                        "The description of a task cannot be empty.");
+            }
+            String taskDescription = taskTokens[1].trim();
+
+            switch (taskType) {
```

Comment:
I do like that for your switch cases, you specified each case instead of "case 1,2,3 etc" as most people do, makes it more readable.